### DISCUSSION TITLE
Is there a way to add GitHub or MS Word level change management?

### DISCUSSION THREAD

**Discussion by jcc5018** [owen-it/laravel-auditing#970] [Q&A] [answered]
I am looking for a change manager similar to GitHub or Word that allows users to accept or reject changes to particular aspects of a text field. 

So instead of recording that the whole field changed, I only want to see that perhaps a user fixed a typo or added a new paragraph. 

Or if that is too complex currently, have a system that highlights changes between old and new. I'm not sure how this would work if you add a sentence in the middle of a paragraph though. Would the whole remaining text be marked as "different" as it is not in the order of letters as in the previous version?

I'm fairly sure GitHub does this by assigning a key that it inserts in the original document that is then referenced and supplemented with the changed value, but I don't believe this package works to that level. Right?

**Comment by parallels999** [comment, marked as answer]
You should do that in your frontend, this would only be the backend.

**Examples:** 
https://neil.fraser.name/software/diff_match_patch/demos/diff.html
https://github.com/liddiard/text-diff
https://www.npmjs.com/package/simple-text-diff
https://jsfiddle.net/ybz8enwk/3/

**Reply by jcc5018** [reply]
Awesome, i think that is just what I need in addition to this auditing package
